### PR TITLE
Satellite specific quenching for satellites

### PR DESCRIPTION
**Pull request by alexalar**
This PR adds satellite-specific quenching for satellite haloes. It does so by adding 4 new parameters for the sigmoid of `frac_quench_sat`, while keeping a separate 4-param sigmoid for the centrals' `frac_quench_cen`. There's a new parameter, `upid` determining if a halo is central or satellite, and `frac_quench` for each halo takes the `frac_quench_cen` or `frac_quench_sat` value depending on `upid`, using `jnp.where`. 

The plots below show results for SMDPL (no-merging). As a result of this change, there's an improvement in the fit quality of `p(sSFR | Mstar)`, for both centrals and satellites, especially for main sequence galaxies. The improvement is not spectacular, but visible if one compares back-and-forth with the model that uses the same frac_quench for both cen and sat. For SMDPL compare to #46. 

One worrisome feature is that the gradient descent gets halted because of a `nan` in the gradients. I suspect the covariance must be hitting some of the boundaries, perhaps because of the `rho_bounds=(-0.3, 0.3)`, which might make the covariance non-positive definite. I want to test what happens if one uses a diagonal covariance.

smhm_logsm
pdf_mstar
pdf_ssfr_centrals
pdf_ssfr_satellites

### CODE REVIEW

**Comment by alexalar** [comment]
Here is how things look like for TNG. To be compared with #47. There is no obvious improvement in my opinion.  

smhm_logsm
pdf_mstar
pdf_ssfr_centrals
pdf_ssfr_satellites

**Comment by alexalar** [comment]
Based on these results, it seems that results improve a bit for SMDPL, but not so clearly for TNG. This might still be useful in the future, but I feel perhaps it is worth testing first a case where the MS parameters are independent for quenched and mainseq galaxies. Potentially setting the off-diagonal covariance to zero, which would compensate the increase of free parameters.

File: diffstarpop/mc_diffstarpop_tpeak_satfrac.py [hunk 1]
```diff
@@ -0,0 +1,462 @@
...
+    frac_q : scalar, float
+        Quenched fraction
+
+    mc_is_q : scalar, bool
+        True for a quenched galaxy and False for unquenched
+
+    """
+    mc_diffstar = mc_diffstar_params_singlegal(
+        diffstarpop_params,
+        logmp0,
```
Comment (alexalar) [inline diffstarpop/mc_diffstarpop_tpeak_satfrac.py:121]: `upid` is missing here
